### PR TITLE
add vjs-waiting while waiting for ad to load

### PR DESCRIPTION
**Pull request by RevinKey**
In VPAID cases the loading spinner does not animate due to vjs-waiting not being added to the player while waiting fro ads to load. reopen of #278

### CODE REVIEW

**Comment by ldayananda** [comment]
@RevinKey It looks like you have a small linting issue that needs to be fixed for the CI tests to run. Do you mind doing that?

**Comment by RevinKey** [comment]
@ldayananda all set it passed CI tests now.

File: src/plugin.js [hunk 1]
```diff
@@ -382,6 +386,8 @@ const contribAdsPlugin = function(options) {
         if (player.ads.nopreroll_) {
           // This will start the ads manager in case there are later ads
           player.trigger('readyforpreroll');
+          // add class for spinner animation while in ?preroll state
+          player.addClass('vjs-waiting');
```
Comment (incompl) [inline src/plugin.js]: If there is no preroll, is it necessary to add vjs-waiting?

Comment (incompl) [reply src/plugin.js]: Ping @RevinKey

**Comment by incompl** [comment]
I made a diagram of where the class is added and removed in your PR and found that it can be simplified.

unnamed

Here is a simplified flow that I recommend based on this:

* Add the class when leaving content-set
* Add the class when entering `postroll?`
* Remove the class on entering `content-playback`
* Remove the class on entering `ad-playback`

**Comment by incompl** [comment]
LGTM

**Comment by sunnyCN34** [comment]
IMA: @brightcove/videojs-ima3@2.20.4-0
@brightcove/videojs-freewheel@2.8.6-1

Issue found:

Animated spinner still visible in background on iPads (IOS 11), when page is loaded and BPB is not pressed.

**Pull request closed without merging**